src/main/java/edu/practise/trip_planner/service/place/PlaceServiceImpl.java

```diff
@@ -0,0 +1,75 @@
+package edu.practise.trip_planner.service.place;
+
+
+import edu.practise.trip_planner.entities.Comment;
+import edu.practise.trip_planner.entities.TourPlace;
+import edu.practise.trip_planner.exceptions.PlaceNotFoundException;
+import edu.practise.trip_planner.repository.CommentsRepository;
+import edu.practise.trip_planner.repository.PlaceRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class PlaceServiceImpl implements PlaceService {
+    private final PlaceRepository placeRepository;
+    private final CommentsRepository commentsRepository;
+
+    public PlaceServiceImpl(PlaceRepository placeRepository, CommentsRepository commentsRepository) {
+        this.placeRepository = placeRepository;
+        this.commentsRepository = commentsRepository;
+    }
+
+    @Override
+    public List<Comment> findCommentsByPlaceId(Long id) {
+        return commentsRepository.findCommentsByPlaceId(id);
+    }
+
+    @Override
+    public Comment addCommentToPlace(Long placeId, Comment comment) {
+        TourPlace place = placeRepository
```

Comment:
Create 2 methods in service:

```
public Place getPlaceById(String id) {
    return findPlaceById(id).orElseThrow("exception message");
}
```

```
public Optional<Place> findPlaceById(String id) {
    return placeRepository.findById(id);
}
```

Reply:
done. Am I understand right, that we call method get when it returns instance of object and find when it returns optional?

Reply:
Yep. Of course, it's up to you but doing it this way makes your code more readable